robottelo/cli/factory.py

```diff
@@ -4240,6 +4241,23 @@ def virt_who_hypervisor_config(
         'organization-id': org['id'],
         'search': 'name={0}'.format(virt_who_hypervisor_hostname)
     })
+    # Note: if one shot command was executed the report is immediately
+    # generated, and the server must have already registered the virt-who
+    # hypervisor host
+    if not org_hosts and not exec_one_shot:
+        # we have to wait until the first report was sent.
+        # the report is generated after the virt-who service startup, but some
+        # small delay can occur.
+        max_time = time.time() + 60
+        while time.time() <= max_time:
+            org_hosts = Host.list({
+                'organization-id': org['id'],
+                'search': 'name={0}'.format(virt_who_hypervisor_hostname)
+            })
+            if org_hosts or time.time() > max_time:
```

Comment:
`time.time() > max_time` looks as redundant

Reply:
No the time was passed we do not need to wait 5 seconds

Reply:
`max_time = time.time() + 55` do the same?

Reply:
@oshtaier think it does not matter how much +- 5 seconds, but for the logic to prevent wasting time at the end, I can put that sleep at the beginning of the loop and remove the time validation, if OK with do that ( will update).

Reply:
yeah, put it into the start of the loop, but it is not so important, just if you like to do that